plugins/woocommerce/src/Internal/Logging/RemoteLogger.php

```diff
@@ -470,6 +485,26 @@ function ( $trace_item ) {
 		return implode( "\n", $sanitized_trace );
 	}
 
+
+	/**
+	 * Redact potential user data from the message.
+	 *
+	 * @param string $message The message to redact.
+	 * @return string The redacted message.
+	 */
+	private function redact_user_data( $message ) {
+		// Redact email addresses.
+		$message = preg_replace( '/[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}/', '[redacted_email]', $message );
+
+		// Redact potential phone numbers (this is a simple pattern and might need refinement).
+		$message = preg_replace( '/\b\d{3}[-.]?\d{3}[-.]?\d{4}\b/', '[redacted_phone]', $message );
```

Comment:
let's make this more generic to include non-US number formats, what do you think of something like
`/\+?(\(?\d{1,3}\)?)?(([-.\s]?\d{2,4}){1,4}|\d{5,10})/` 

Explanation

	1.	\+?: Matches an optional + at the start.
	2.	(\(?\d{1,3}\)?)?: Matches an optional area code, which:
	•	\(?: Allows an optional opening parenthesis (.
	•	\d{1,3}: Matches 1 to 3 digits for the area code.
	•	\)?: Allows an optional closing parenthesis ).
	3.	(([-.\s]?\d{2,4}){1,4}|\d{5,10}): Matches either:
	•	([-.\s]?\d{2,4}){1,4}: 1 to 4 groups, each containing 2 to 4 digits, optionally separated by -, ., or whitespace.
	•	|: OR
	•	\d{5,10}: A consecutive string of 5 to 10 digits.

Example Matches

This regex will match patterns like:

	•	+1 (123) 456 7890 (with parentheses around area code)
	•	+44-123-4567-890 (with area code, no parentheses)
	•	1234567890 (10 consecutive digits, no area code)
	•	(123) 456-7890 (area code in parentheses, groups)
	•	+91 12345 (5 consecutive digits, no parentheses)


Phone number match is [notoriously difficult](https://github.com/google/libphonenumber), but this is a good start and thanks for making the effort!

Reply:
Thanks for the suggestion!  I'll let you know once I've made the changes.

Reply:
I've updated the regex to include more number formats and added more test cases to ensure it covers a wide range of phone number variations. 0100bf4a4f8bd890a2929386e7acdbe8d2a03c6e

I didn't use the provided regex because it also matches the number like `123`. I asked ChatGPT for confirmation on this.